### PR TITLE
[CDAP-18290] Catch ArrayIndexOutOfBounds in PipelineSpecGenerator

### PR DESCRIPTION
**Pull request by ghost**
[CDAP-18290 - ValidationException should log stacktraces](https://cdap.atlassian.net/browse/CDAP-18290)

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/spec/PipelineSpecGenerator.java [hunk 1]
```diff
@@ -348,6 +348,13 @@ protected ConfiguredStage configureStage(ETLStage stage, ValidatedPipeline valid
         .withStacktrace(e.getStackTrace());
       // Log the NullPointerException for debugging:
       LOG.error(String.format("Null error occurred while configuring the stage %s.", stageName), e);
+    } catch (ArrayIndexOutOfBoundsException e) {
+      // handle the case where plugin throws array index out of bounds exception,
+      // this is to avoid having a number like '2', '8' etc as error message
+      collector.addFailure(String.format("ArrayIndexOutOfBounds error occurred while configuring the stage %s.",
```
Comment (CuriousVini) [inline cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/spec/PipelineSpecGenerator.java]: Lets say `Index out of bounds ..`

**Review by CuriousVini — APPROVED** [review]
single minor comment, LGTM 👍🏼

File: cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/spec/PipelineSpecGenerator.java (hunk 1 above)
Comment (ghost) [reply cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/spec/PipelineSpecGenerator.java]: Changed

File: cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/spec/PipelineSpecGenerator.java [hunk 2]
```diff
@@ -348,6 +348,13 @@ protected ConfiguredStage configureStage(ETLStage stage, ValidatedPipeline valid
         .withStacktrace(e.getStackTrace());
       // Log the NullPointerException for debugging:
       LOG.error(String.format("Null error occurred while configuring the stage %s.", stageName), e);
+    } catch (ArrayIndexOutOfBoundsException e) {
+      // handle the case where plugin throws index out of bounds exception,
+      // this is to avoid having a number like '2', '8' etc as error message
+      collector.addFailure(String.format("Index out of bounds error occurred while configuring the stage %s.",
+                                         stageName), null).withStacktrace(e.getStackTrace());
+      // Log the ArrayIndexOutOfBoundsException for debugging:
+      LOG.error(String.format("ArrayIndexOutOfBounds error occurred while configuring the stage %s.", stageName), e);
```
Comment (chtyim) [inline cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/spec/PipelineSpecGenerator.java]: Why log? the error is already responded back to the client and it is a client side error?

Comment (ghost) [reply cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/spec/PipelineSpecGenerator.java]: I was referring to the way Null Pointer Exceptions are caught 1 line above this. Let me know if it needs to be removed.

Comment (chtyim) [reply cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/spec/PipelineSpecGenerator.java]: I think we shouldn't log this.

Comment (ghost) [reply cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/spec/PipelineSpecGenerator.java]: Removed

**Comment by ghost** [comment]
Will fix failing tests and merge on Monday, 13th September.

**Pull request merged**